### DISCUSSION TITLE
Modbus

### DISCUSSION THREAD

**Discussion by StefanoPirazzi** [indomus/forum#1589] [HUB: Home Assistant]
Ho appena trasferito tutto da HA core in ambiente Vem a HomeassistantOS (HASSIO) ver. core-2022.3.5

Purtroppo non mi legge più i valori del mio inverter Huawai SUN2000 tramite  Modbus con collegamento tramite IP:

modbus:
  name: solar
  type: tcp
  host: 192.168.xxx.xxx
  port: 502
  delay: 10
  close_comm_on_error: true
  retry_on_empty: true
  retries: 10

Ho provato a cercare su internet, pensando che fosse cambiato qualcosa a livollo di impostazioni, ma non ho trovato nulla.

Ho installato per provare l' Add-on tramite hacs "Huawei Solar" ma purtroppo mi restituisce il seguente errore:

Config entry 'SUN2000L-5KTL' for huawei_solar integration not ready yet: Could not update 2101073796HVHB005492 values: Got error while reading from register 37760 with length 28: Exception Response(131, 3, IllegalAddress); Retrying in background

Come posso tornare a leggere i miei sensori dell'inverter?

Grazie

**Comment by vinny74man** [comment]
Scusa ma l'installazione è su docker?
Nel caso controlla le impostazioni di rete che hai configurato su docker, poi la configurazione che hai postato è solo la connessione al dispositivo, ma non  vedo i sensori con l'eventuale id e i registri, probabilmente avevi una vecchia versione prima e magari nel frattempo hanno cambiato un po' di cose, comunque se cerchi nelle pagine ufficiali di HA il modbus è ben documentato, io non avendolo mai fatto ho inserito tutte le letture di energia di una fabbrica con relativo consumo energetico e produzione solare.

**Comment by StefanoPirazzi** [comment]
No non é su docker é direttamente Hassio su Raspberry. Si, è solo la parte
di impostazione modbus, poi ho tutti i sensori in un file yaml nella
cartella sensors. É la stessa configurazione che avevo su homesassistant
core in ambiente venv che non aggiornava da un anno per problemi in fase di
aggiornamento, ed ho deciso di passare ad Hassio ultima versione. Funziona
tutto tranne questo.


Il lun 21 mar 2022, 18:19 vinny74man ***@***.***> ha scritto:

**Reply by vinny74man** [reply]
dai log non riesci a capire cosa succede, se magari non è raggiungibile l'inverter o qualche sensore ritorna qualche dato strano, io ho aggiornato l'altro ieri sera alla versione 2022.3.5 e tutti i sensori modbus funzionano benissimo.
Prova a controllare i log per cercare di capire cosa succede

**Reply by StefanoPirazzi** [reply]
Ho un po di casino con i binary_sensor tipo template, mi danno errore e warning, è tutta mattina che guardo su internet, ma non ci cavo i piedi. Non avevo questi problemi prima ....

Logger: homeassistant.helpers.template
Source: helpers/template.py:1286
First occurred: 10:18:10 (75 occurrences)
Last logged: 10:19:00

Template warning: 'int' got invalid input 'unknown' when rendering template '{{ states.sensor.asciugatrice_istant.state | int > 1500 }}' but no default was specified. Currently 'int' will return '0', however this template will fail to render in Home Assistant core 2022.1
Template warning: 'int' got invalid input 'unknown' when rendering template '{{ states.sensor.asciugatrice_istant.state | int < 1500 and states.sensor.asciugatrice_istant.state | int > 300 }}' but no default was specified. Currently 'int' will return '0', however this template will fail to render in Home Assistant core 2022.1
Template warning: 'int' got invalid input 'unknown' when rendering template '{{ states.sensor.asciugatrice_istant.state | int < 300 and states.sensor.asciugatrice_istant.state | int > 6 }}' but no default was specified. Currently 'int' will return '0', however this template will fail to render in Home Assistant core 2022.1
Template warning: 'float' got invalid input 'unknown' when rendering template '{{ (( states.sensor.clima_giorno_shellyem_clima.state | float ) + ( states.sensor.clima_notte_shellyem_clima.state | float )) | round(3) }}' but no default was specified. Currently 'float' will return '0', however this template will fail to render in Home Assistant core 2022.1
Template warning: 'float' got invalid input 'unknown' when rendering template '{{ (( states.sensor.fv_shellyem_energia.state | float ) + ( states.sensor.enel_shellyem_energia.state | float ) - ( states.sensor.enel_returned_shellyem_energia.state | float )) | round(3) }}' but no default was specified. Currently 'float' will return '0', however this template will fail to render in Home Assistant core 2022.1

I sensori esistono ed hanno anche valori

E' cambiato il modo di scrivere i binary_sensor?
Questi sono dentro un file binary_sensor.yaml

- platform: template
  sensors:
    stato_operativo_lavatrice:
      friendly_name: "Stato Operativo Lavatrice"
      value_template: '{{ states.sensor.lavatrice_istant.state | int > 6 }}'
      device_class: power
      delay_on:
        seconds: 30
      delay_off:
        minutes: 2

- platform: template
  sensors:
    riscaldamento_lavatrice:
      value_template: '{{ states.sensor.lavatrice_istant.state | int > 1500 }}'
      delay_on:
        minutes: 2
      delay_off:
        minutes: 2
    lavaggio_lavatrice:
      value_template: "{{ states.sensor.lavatrice_istant.state | int < 1500 and states.sensor.lavatrice_istant.state | int > 300 }}"
      delay_on:
        minutes: 2
      delay_off:
        minutes: 2
    risciacquo_lavatrice:
      value_template: "{{ states.sensor.lavatrice_istant.state | int < 300 and states.sensor.lavatrice_istant.state | int > 6 }}"
      delay_on:
        minutes: 2
      delay_off:
        minutes: 2

**Reply by vinny74man** [reply]
Questi warning li avevo anchio ieri ho trovato un post con il quale li ho risolti, dalla versione 2022.1, mi sembra, è cambiato qualcosa che adesso non ricordo, comunque quando si riavvia HA finchè non viene letto il valore dei sensori modbus questi hanno come valore unknown per risolvere almeno per me che uso dei sensori che hanno dei valori float basta aggiungere il valore di default tra parentesi dopo il tipo, nel tuo caso questo
`value_template: '{{ states.sensor.lavatrice_istant.state | int > 6 }}'` dovrebbe essere così
`value_template: '{{ states.sensor.lavatrice_istant.state | int(0) > 6 }}'` se zero è il tuo valore di default, in questo modo io ho risolto non ho più avuto questo tipo di problema

**Reply by vinny74man** [reply]
ovviamente per tutti i sensori devi fare la stessa cosa

**Comment by StefanoPirazzi** [comment]
Si, lo ho trovato anche io oggi int(default=0) se ricordo bene, sei molto
gentile, grazie per il momento, poi ti aggiorno!

Il mar 22 mar 2022, 20:51 vinny74man ***@***.***> ha scritto:
